### PR TITLE
Add Raises comments to samplers

### PR DESCRIPTION
**Pull request by yuk1ty**
## Motivation

- See issue #1618 for the original motivation.

## Description of the changes

Another addition of `Raises:` comments. I worked on `samplers` module as far as I found.

### CODE REVIEW

File: optuna/samplers/_search_space.py [hunk 1]
```diff
@@ -43,6 +43,10 @@ def calculate(
 
         Returns:
             A dictionary containing the parameter names and parameter's distributions.
+
+        Raises:
+            ValueError:
+                If ``self._study_id`` is not equal to ``study._study_id``.
```
Comment (toshihikoyanase) [inline optuna/samplers/_search_space.py:49]: I guess this sentence could be more intuitive. How about rephrasing it by using the expression in `ValueError` at L54? The following is just an example:

> If multiple studies are inputted into this method.

**Review by toshihikoyanase — CHANGES_REQUESTED** [review]
Thank you for your PR as always. I have some small comments.

**Comment by toshihikoyanase** [comment]
Let me replace the `code-fix` label with the `document` label since this PR only updates docstring.

File: optuna/samplers/_search_space.py (hunk 1 above)
Comment (hvy) [reply optuna/samplers/_search_space.py:49]: Definitely a nit but "different" might be clearer than "multiple". It's also what we use for describing this behavior in the docs of `__init__`.

> If different studies are passed ..

**Comment by hvy** [comment]
Looks great to me besides the nit above and @toshihikoyanase's comments.

File: optuna/samplers/_search_space.py (hunk 1 above)
Comment (toshihikoyanase) [reply optuna/samplers/_search_space.py:49]: > If different studies are passed ..

@hvy Much better! Thank you for your suggestion.

**Review by hvy — APPROVED** [review]
LGTM!

File: optuna/samplers/_search_space.py (hunk 1 above)
Comment (yuk1ty) [reply optuna/samplers/_search_space.py]: More clear than before. Thank you!

File: optuna/samplers/_cmaes.py [hunk 2]
```diff
@@ -142,6 +142,10 @@ def objective(trial):
                 to set this flag :obj:`True` when the :class:`~optuna.pruners.HyperbandPruner` is
                 used. Please see `the benchmark result
                 <https://github.com/optuna/optuna/pull/1229>`_ for the details.
+
+    Raises:
+        ValueError:
+            If ``restart_strategy`` is not 'ipop' or :obj: `None`.
```
Comment (toshihikoyanase) [inline optuna/samplers/_cmaes.py:148]: A nit, but could you remove a white space between `:obj:` and `None`?
```suggestion
            If ``restart_strategy`` is not 'ipop' or :obj:`None`.
```

image

**Review by toshihikoyanase — CHANGES_REQUESTED** [review]
I found a minor error, so could you take a look?

File: optuna/samplers/_cmaes.py (hunk 2 above)
Comment (yuk1ty) [reply optuna/samplers/_cmaes.py]: oh, sorry, thank you!

Comment (hvy) [reply optuna/samplers/_cmaes.py]: Sorry should've noticed as well. Thanks @toshihikoyanase and @yuk1ty .

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your swift action. LGTM!

File: optuna/samplers/_cmaes.py (hunk 2 above)
Comment (toshihikoyanase) [reply optuna/samplers/_cmaes.py]: No problem at all. We employ two-approval rule to help each other.

**Pull request merged**